src/pgsql.c

```diff
@@ -75,6 +77,39 @@ kore_pgsql_init(void)
 }
 
 int
+kore_pgsql_init_sync(struct kore_pgsql *pgsql)
+{
+	struct pgsql_conn *conn;
+
+	pgsql->state = KORE_PGSQL_STATE_INIT;
+	pgsql->result = NULL;
+	pgsql->error = NULL;
+	pgsql->conn = NULL;
+
+	if (TAILQ_EMPTY(&pgsql_conn_free)) {
+		if (pgsql_conn_count >= pgsql_conn_max) {
+			pgsql->error = kore_strdup(PGSQL_ERROR_MAX_CONN);
+			return (KORE_RESULT_ERROR);
+		}
+
+		if (!pgsql_conn_create(pgsql)) {
+			pgsql->error = kore_strdup(PGSQL_ERROR_CONNECTING);
```

Comment:
Same as below, pgsql_conn_create() will set its own pgsql->error value if it fails. So the assignment can be removed as well here.